### ISSUE TITLE
Weird communication results 

### ISSUE THREAD

**Issue by khoaguin** [snwagh/falcon-public#45]
Hi, I am training LeNet on localhost for 15 iterations. I got the results like in the picture, where it shows the total communication is 900 Mb, but the communication for P0 is already 1673.28 MB. Is there anything wrong with this?  

![image](https://user-images.githubusercontent.com/88959106/177245100-9fad4321-f4c7-445a-b0e9-d454f4c16664.png)

**Comment by snwagh** [comment]
I think there is some issue with the communication part (there is some issue with parallelization as well). Either it is wrapping around the bit size or something else is off. Can you please probe into this issue further?

**Comment by khoaguin** [comment]
Yes, I am looking into it. So for example, if I change the number of iterations to 1, the communications become normal again
![image](https://user-images.githubusercontent.com/88959106/177508469-4ec906a7-7ea6-4922-bbe9-5cc6409bf508.png)
